libclc/ptx-nvidiacl/libspirv/relational/isnan.cl

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+
+#include "../../include/libdevice.h"
+#include <relational.h>
+
+_CLC_DEF _CLC_OVERLOAD bool __spirv_IsNan(float x) { return __nv_isnanf(x); }
```

Comment:
Does it enable the test recently updated by Bjoern: https://github.com/intel/llvm/commit/7c15695e477fcf91e37eeadbb5a0aa9ba17b88ed?
If so, we should remove XFAIL from the test to avoid "unexpected passes".

Reply:
No, this PR is about performance and using more native function for CUDA HW where applicable.

`nan.cpp` still fails because of half (enabled in anther PR)

Reply:
Thanks for clarification.